src/quilt/portablemc_quilt/__init__.py

```diff
@@ -0,0 +1,164 @@
+from argparse import ArgumentParser, Namespace
+from typing import Optional
+import sys
+
+from portablemc import Version, BaseError, Context, JsonRequestError, json_simple_request
+
+
+QUILT_META_URL = "https://meta.quiltmc.org/{}"
+QUILT_VERSIONS_LOADER = "v3/versions/loader/{}"
+QUILT_VERSIONS_LOADER_PROFILE = "v3/versions/loader/{}/{}/profile/json"
+
+
+def load():
+
+    from portablemc.cli import CliContext
+    from portablemc import cli as pmc
+
+    # Private mixins
+
+    @pmc.mixin()
+    def register_start_arguments(old, parser: ArgumentParser):
+        _ = pmc.get_message
+        parser.add_argument("--quilt-prefix", help=_("args.start.quilt_prefix"), default="quilt", metavar="PREFIX")
+        old(parser)
+
+    @pmc.mixin()
+    def cmd_start(old, ns: Namespace, ctx: CliContext):
+        try:
+            return old(ns, ctx)
+        except QuiltInvalidFormatError:
+            pmc.print_task("FAILED", "start.quilt.error.invalid_format", done=True)
+            sys.exit(pmc.EXIT_WRONG_USAGE)
+        except QuiltVersionNotFound as err:
+            pmc.print_task("FAILED", f"start.quilt.error.{err.code}", {"version": err.version}, done=True)
+            sys.exit(pmc.EXIT_VERSION_NOT_FOUND)
+
+    @pmc.mixin()
+    def new_version(old, ctx: CliContext, version_id: str) -> Version:
+
+        if version_id.startswith("quilt:"):
+
+            version_split = version_id.split(":")
+            if len(version_split) > 3:
+                raise QuiltInvalidFormatError()
+
+            game_version = version_split[1]
+            loader_version = version_split[2] if len(version_split) == 3 else None
+
+            if not len(game_version):
+                game_version = "release"
+
+            if loader_version is not None and not len(loader_version):
+                raise QuiltInvalidFormatError()
+
+            manifest = pmc.new_version_manifest(ctx)
+            game_version, _game_version_alias = manifest.filter_latest(game_version)
+
+            if loader_version is None:
+                pmc.print_task("OK", "start.quilt.resolving_loader", {
+                    "game_version": game_version
+                }, done=True)
+            else:
+                pmc.print_task("OK", "start.quilt.resolving_loader_specific", {
+                    "loader_version": loader_version,
+                    "game_version": game_version
+                }, done=True)
+
+            ver = QuiltVersion(ctx, game_version, loader_version, prefix=ctx.ns.fabric_prefix)
```

Comment:
Use the `quilt_prefix` command line argument's value instead of fabric's one

Reply:
updated

Reply:
updated